Fetch Android dependencies from Github releases

dylanhand opened the pull request:
Updates Android to download `divvunspell` and `pahkat` dependencies from Github releases instead of using pahkat prefix.

bbqsrc commented:
Closed in favour of https://github.com/divvun/kbdgen/commit/ebc617883699c802f407f9401c1efbc5b21a6f98 and https://github.com/divvun/kbdgen/commit/e1d02c6fe27a8805bea156cf7e940d5e0124ae95. Thanks!

dylanhand committed:
Begin implementation of fetching Android deps from github releases

dylanhand committed:
it works but it's ugly

dylanhand committed:
Simplify kinda

dylanhand committed:
Checkout repo first, download deps directly

Committing because it works. Will clean this up to use system tar now that divvunspell packages jniLibs like pahkat.

dylanhand committed:
Extract with system tar

dylanhand committed:
Download directly to file

dylanhand committed:
Unused imports

dylanhand committed:
Fix extra jniLibs path after extraction

dylanhand committed:
Use .expect for errors

dylanhand committed:
Use tracing for logging

dylanhand committed:
Remove comment

dylanhand committed:
Make explicit this is android deps, not github

dylanhand committed:
ok cargo fmt

dylanhand committed:
Remove dead pahkat-based android deps code

dylanhand committed:
Test no kbdgen deploy on branches
